src/event/block/BlockExplodeEvent.php

```diff
@@ -53,11 +52,13 @@ public function __construct(
 	){
 		parent::__construct($block);
 
+		Utils::checkFloatNotInfOrNaN("fireChance", $fireChance);
+		Utils::checkFloatNotInfOrNaN("yield", $yield);
 		if($yield < 0.0 || $yield > 100.0){
 			throw new \InvalidArgumentException("Yield must be in range 0.0 - 100.0");
 		}
 		if($fireChance < 0.0 || $fireChance > 1.0){
-			throw new \InvalidArgumentException("Fire chance must be between 0 and 1.");
+			throw new \InvalidArgumentException("Fire chance must be a infinite or nan number between 0 and 1.");
```

Comment:
You don't ned to mention infinite or nan here, it's already reported by Utils if needed